static-site/src/components/ListResources/IndividualResource.tsx

```diff
@@ -111,20 +129,20 @@ export const IndividualResource = ({data, isMobile}) => {
 
       <FlexRow>
 
-        <TooltipWrapper description={`Last known update on ${data.lastUpdated}`}>
-          <ResourceLinkWrapper onShiftClick={() => setModal(data)}>
-            <Name displayName={data.displayName} href={data.url} topOfColumn={topOfColumn}/>
+        <TooltipWrapper description={`Last known update on ${resource.lastUpdated}`}>
+          <ResourceLinkWrapper onShiftClick={() => setModalResource && setModalResource(resource as VersionedResource)}>
+            <Name displayName={resource.displayName} href={resource.url} topOfColumn={topOfColumn}/>
           </ResourceLinkWrapper>
         </TooltipWrapper>
 
-        {data.versioned && !isMobile && (
-          <TooltipWrapper description={data.updateCadence.description +
-            `<br/>Last known update on ${data.lastUpdated}` +
-            `<br/>${data.nVersions} snapshots of this dataset available (click to see them)`}>
+        {resource.versioned && !isMobile && (
+          <TooltipWrapper description={(resource as VersionedResource).updateCadence.description +
+            `<br/>Last known update on ${resource.lastUpdated}` +
+            `<br/>${resource.nVersions} snapshots of this dataset available (click to see them)`}>
             <IconContainer
               Icon={MdHistory}
-              text={`${data.updateCadence.summary} (n=${data.nVersions})`}
-              handleClick={() => setModal(data)}
+              text={`${(resource as VersionedResource).updateCadence.summary} (n=${resource.nVersions})`}
+              handleClick={() => setModalResource && setModalResource(resource as VersionedResource)}
```

Comment:
Stylistic suggestion:

To increase signal-to-noise ratio, you could use JavaScript's Elvis operator ([optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)). It works for function calls too:

```ts
setModalResource?.(resource)
```

Interestingly, the null check was not needed in js version. It's either that the function (context) can be typed more strictly (i.e. the function actually cannot be nil) or that this has been a bug in js version. To know, need to check context docs and how it is initialized in this particular case. But I think this does not matter much to be worth a hassle.

Reply:
Done:

https://github.com/nextstrain/nextstrain.org/blob/adc666a4ce56df6011a2ad3c076406512810bf38/static-site/src/components/ListResources/IndividualResource.tsx#L146-L147

Reply:
TypeScript is correct and it's possible for `setModalResource` to be `null` if the `SetModalResourceContext.Provider` parent is removed. The child component has no way of knowing if it's there or not.

https://github.com/nextstrain/nextstrain.org/blob/adc666a4ce56df6011a2ad3c076406512810bf38/static-site/src/components/ListResources/index.tsx#L77-L78

Such a case would definitely be an internal error though, not user error. I've updated to raise an error in this case (see [force-push](https://github.com/nextstrain/nextstrain.org/compare/adc666a4ce56df6011a2ad3c076406512810bf38..2ad11e7bc97f3a54c4a819096b9a270adfc58eb7)).